src/error_chain.rs

```diff
@@ -137,6 +137,17 @@ macro_rules! error_chain_processed {
                 )
             }
 
+            /// Construct a chained error from another boxed error and a kind, and generates a backtrace
+            pub fn with_boxed_chain<K>(error: Box<::std::error::Error + Send>, kind: K)
+                -> $error_name
+                where K: Into<$error_kind_name>
+            {
+                $error_name(
+                    kind.into(),
+                    $crate::State::new::<$error_name>(error, ),
+                )
+            }
+
```

Comment:
Yes, I tried that as well. Unfortunately, the problem we'd need to express is "if you have a boxed trait object, use the box as-is, and if you have an unboxed error object, box it". The problem is that because we have `impl Error for Box<Error>`, the compiler doesn't do the right thing since boxed errors look like unboxed errors.